File: power_of_thor/taille_min/pot_taille.java
package taille_min;
import java.util.*;
class Player{public static void main(String a[]){
Scanner s=new Scanner(System.in);
int lx=s.nextInt(),ly=s.nextInt(),tx=s.nextInt(),ty=s.nextInt();
while(s.hasNext()){s.next();int dx=Integer.compare(lx,tx),dy=Integer.compare(ly,ty);
tx+=dx;ty+=dy;System.out.println((dy>0?"S":dy<0?"N":"")+(dx>0?"E":dx<0?"W":""));}
s.close();}}

File: power_of_thor/taille_min/pot_taille.kt
import java.util.*
fun main() {
  val i = Scanner(System.`in`)
  val lx = i.nextInt()
  val ly = i.nextInt()
  var tx = i.nextInt()
  var ty = i.nextInt()
  while (true){
    i.nextInt()
    print((if(ty<ly)"S" else if(ty>ly)"N" else "").also{ty+=ly.compareTo(ty)}+(if(tx<lx)"E" else if(tx>lx)"W" else "").also{tx+=lx.compareTo(tx)}+"\n")}
}
